build_mt.py

```diff
@@ -1,599 +1,827 @@
+            )
+
+            ax = plt.gca()
+            ax.cla()  # clear things for fresh plot
+
+            # ax.scatter(pos_this_info[choice_lc,1],pos_this_info[choice_lc,2],s=0.01,alpha=0.5,color='orange')
+            ax.scatter(
+                pos_choice[choice, 1],
+                pos_choice[choice, 2],
+                s=0.01,
+                alpha=0.5,
+                color="dodgerblue",
+            )
+
+            # circles for in and prev
+            ax.add_artist(circle_this)
+            ax.add_artist(circle_prev)
+            plt.xlabel([-1000, 3000])
+            plt.ylabel([-1000, 3000])
+            plt.axis("equal")
+            plt.show()
+
+        del (
+            com_dist_this,
+            main_prog_this,
+            halo_ind_this,
+            pos_this,
+            N_halos_slabs_this,
+            slabs_this,
+        )
+        del (
+            com_dist_prev,
+            main_prog_prev,
+            halo_ind_prev,
+            pos_prev,
+            N_halos_slabs_prev,
+            slabs_prev,
+        )
+        gc.collect()
+
+        # update values for difference in comoving distance, eligibility and number of copies
+        delta_chi_old = delta_chi
+        eligibility_this = eligibility_prev
+        copies_old = copies_prev
+
+        # save the current state so you can resume
+        np.save(cat_lc_dir / "tmp" / "eligibility_prev.npy", eligibility_prev)
+        np.save(cat_lc_dir / "tmp" / "halo_ind_next.npy", halo_ind_next)
+        np.save(cat_lc_dir / "tmp" / "pos_star_next.npy", pos_star_next)
+        np.save(cat_lc_dir / "tmp" / "vel_star_next.npy", vel_star_next)
+        np.save(cat_lc_dir / "tmp" / "chi_star_next.npy", chi_star_next)
+        np.save(cat_lc_dir / "tmp" / "z_prev_delta_copies.npy",
+            np.array([z_prev, delta_chi, copies_prev]),
+        )
+
+    # dict_keys(['HaloIndex', 'HaloMass', 'HaloVmax', 'IsAssociated', 'IsPotentialSplit', 'MainProgenitor', 'MainProgenitorFrac', 'MainProgenitorPrec', 'MainProgenitorPrecFrac', 'NumProgenitors', 'Position', 'Progenitors'])
+    
+
+class ArgParseFormatter(argparse.RawDescriptionHelpFormatter, argparse.ArgumentDefaultsHelpFormatter):
+    pass
+
+
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=ArgParseFormatter)
+    parser.add_argument('sim_name', help='Simulation name', default=DEFAULTS['sim_name'])
+    parser.add_argument('z_start', help='Initial redshift where we start building the trees', default=DEFAULTS['z_start'])
+    parser.add_argument('z_stop', help='Final redshift (inclusive)', default=DEFAULTS['z_end'])
+    parser.add_argument('--resume', help='Resume the calculation from the checkpoint on disk', action='store_true')
+    parser.add_argument('--plot', help='Want to show plots', action='store_true')
+    
+    args = vars(parser.parse_args())
+    
+    main(**args)
```

Comment:
argparse command-line interfaces are the best!  I find them super quick to write and they serve as both documentation and user interface.  If you can name the args the same as the `main` args, then running the program is as quick as `main(**args)`.

Reply:
And the defaults can be set from the global values so you don't have to save a long command line invocation when testing.

Reply:
a big fan of this! Thank you!